metagraph/src/annotation/annotation_converters.cpp

```diff
@@ -991,22 +991,26 @@ std::unique_ptr<RowDiffAnnotator> convert_to_row_diff(const graph::DBGSuccinct &
     uint64_t nnodes = graph.num_nodes();
     Vector<uint64_t> node_diffs;
     Vector<std::pair<uint64_t, uint32_t>> pos(nnodes, { 0, 0 });
-    sdsl::bit_vector terminal(nnodes, 0);
+    sdsl::bit_vector terminal;
 
     std::atomic<uint64_t> terminal_count = 0;
     std::atomic<uint64_t> forced_terminal_count = 0;
     std::atomic<uint64_t> depth_terminal_count = 0;
     std::atomic<uint64_t> boundary_size = 0;
     std::atomic<uint64_t> visited_nodes = 0;
-    graph.get_boss().call_sequences(
-        [&](const std::string &, const std::vector<uint64_t> &path) {
+    graph.get_boss().call_sequences_row_diff(
+        [&](const std::vector<uint64_t> &path, std::optional<uint64_t> anchor) {
```

Comment:
It would be good to build the bitmap `terminal` in a separate function, which would be called here and in the other (more scalable) construction algorithm.

Reply:
In order to build `terminal` we need to traverse the graph. In the more scalable construction algorithm (in branch `rd_brwt`) i am simply calling this function with an empty callback, which is convenient. Are you worried about the overhead of building a path that is not used?